### DISCUSSION TITLE
SyntaxError: Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by scr6w** [community/community#130799] [Programming Help] [answered]
### Body

**Basically I have an html and two js files and from this html file I call one of these js files:**
`<script src="script.js"></script>`
**in script.js I import a hello world function from another file but for some reason the error comes out:**
`Cannot use import statement outside a module`
script.js: 
```
import { sayHelloWorld } from './hello.js';
sayHelloWorld();
```
hello.js:
```
const sayHelloWorld = () => {
    console.log('Hello World');
    };
export { sayHelloWorld };
```

Directory:
![image](https://github.com/community/community/assets/159715242/7e60abc4-971e-4fa9-b900-4dd60b5548b8)

Help!!!!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by diaslui** [comment, marked as answer]
This problem is caused by the fact that JavaScript modules require a specific way to be loaded into HTML files. To use import and export statements, you need to ensure that your script is treated as a module. You can do this by adding the type="module" attribute to your <script> tag in the HTML file.

**So when importing the script into HTML it would look like this👍:**
`<script type="module" src="script.js"></script>`

**Result:**
![image](https://github.com/community/community/assets/115284250/6ea5ac81-8329-475b-9a1f-ee054d262cc2)


It even seems like this has already been discussed here on the github community

**Reply by PPQAQ** [reply]
Can i ask a question? i used  type="module",but it cannot import png,audio or css,do you know how to solve it? appreciate you！
![901720170731_ pic](https://github.com/community/community/assets/117715297/49994eab-6ac6-40ec-9f0b-03e8aded5f83)

**Reply by diaslui** [reply]
I had never come across this, but I managed to reproduce the error
> (let me know if what you did to receive the error was different)

From what I understand, you try to import an image within a script that has type="module"
`import { z } from "./z.jpg" // try import a jpg image
`
but unfortunately this cannot be done because the import is reserved for only js scripts and not other files

so the image must be imported in other ways, I will leave a solution below
```
const img = new Image()
img.src = "./z.jpg" // here we can use the image however we want
```
